tiledb/ml/models/base.py

```diff
@@ -73,38 +104,37 @@ def preview(self, **kwargs):
         PyTorch etc.
         """
 
-    def set_file_properties(
-        self, framework: str, framework_version: str, preview: str
-    ) -> dict:
+    def set_file_properties(self):
         """
-        Abstract method that returns model array's file properties.
-        :param framework: str. Machine learning framework.
-        :param framework_version: str. Machine learning framework's version.
-        :param preview: str. Machine learning model's preview.
-        :return: dict. Dictionary with model's file properties
+        Method that sets model array's file properties.
         """
-        return {
-            self.file_properties_obj.ML_FRAMEWORK.value: framework,
-            self.file_properties_obj.STAGE.value: "STAGING",
-            self.file_properties_obj.PYTHON_VERSION.value: platform.python_version(),
-            self.file_properties_obj.FRAMEWORK_VERSION.value: framework_version,
-            self.file_properties_obj.MODEL_PREVIEW.value: preview,
+        self._file_properties = {
+            ModelFileProperties.ML_FRAMEWORK.value: self.framework,
+            ModelFileProperties.ML_FRAMEWORK_VERSION.value: self.framework_version,
+            ModelFileProperties.STAGE.value: "STAGING",
+            ModelFileProperties.PYTHON_VERSION.value: platform.python_version(),
+            ModelFileProperties.PREVIEW.value: self.preview(),
```

Comment:
Why not set this in `__init__`? Can any of these values change after initialization?

Reply:
I had the same thought, but I decided to put it in a setter function because this could change in the future.